first-drafts (#1)
tesseractcollective/react-graphql-docs: README.md

# What is this and why do I care?

> `react-graphql` is a library built to make CRUD easy when working with graphql back-ends that standardize their format.  We've primarily built it using Hasura.io and use their naming convention as our defaults.

Reviewer: ```suggestion
`react-graphql` is a library built to make CRUD easy when working with graphql back-ends that standardize their format.  We've primarily built it using [Hasura.io](https://hasura.io) and use their naming convention as our defaults.
```

> RG primarily uses hooks to provide support for query one, query many, mutation one, delete one, insert one.

Reviewer: is `RG` the "official" shorthand for `react-graphql`?

> Developer experience is going to involve one time setup of your code generator and one time definition of your typeNames/resources and then the ability to pass that resource config around to the various hooks to get easy typed graphql operations.

Reviewer: Pedantry for compound adjectives and one minor phrase adjustment
```suggestion
Developer experience is going to involve one-time setup of your code generator and one-time definition of your typeNames/resources and then the ability to pass that resource config around to the various hooks to get typed graphql operations easily.
```

> `react-graphql-ui` is a library built on top of react-graphql to give you various components and hooks to help speed up development.  It's main offerings are currently: 

Reviewer: ```suggestion
`react-graphql-ui` is a library built on top of react-graphql to give you various components and hooks to help speed up development.  Its main offerings are currently: 
```

* useDataTable: a hook helping you quickly compose a DataTable to your liking with full pagination, filtering, ordering, and more

> * FlexForm/FlexFormLocal: A component that will build a customizable form, complete with UI components, for you based on your graphql fragment/config.

Reviewer: Match other bullets
```suggestion
* FlexForm/FlexFormLocal: a component that will build a customizable form, complete with UI components, for you based on your graphql fragment/config.
```

> * SelectViaRelationship: a select box that pulls from your DB for it's options, 

Reviewer: Could maybe use more selling/explanation? But I'm sure someone who wants it would dig into the other docs
```suggestion
* SelectViaRelationship: a select box component that pulls from your DB for its options
```

# Initial Setup

## Dependencies 
Install these dependencies:

> `@tesseractcollective/react-graphql primereact primeicons primeflex react-hook-form jotai urql`

Reviewer: Are these required even if the project doesn't use them directly? I think so due to, what are they called, peer dependencies?

Is it possible to make the dependencies that the user may not need to use directly like `urql` or even `react-hook-form` internal dependencies somehow?

Author: Yes, but with some further abstraction.  As it is now, if you aren't using Urql you probably aren't using react-graphql.  And if you aren't using react-hook-form you probably aren't using react-graphql-ui because their utility without those two libraries is minimal.

> `@tesseractcollective/react-graphql primereact primeicons primeflex react-hook-form jotai urql`

Reviewer 2: can we list them in an ordered list, just for clarity sake

Author: No, They are like this so you can easily copy/paste them onto your command line.

And these dev dependencies:

> `postcss-import @graphql-codegen/cli @graphql-codegen/introspection @graphql-codegen/typescript @graphql-codegen/typescript-operations postcss postcss-nested`

Reviewer 2: same here

## react-graphql-ui Provider

> ```

Reviewer: ```suggestion
```ts
```

import { ReactGraphqlUIContext, PrimeReactDefaultComponents } from '@tesseractcollective/react-graphql-ui';
import HasuraConfig from './...../HasuraConfig.ts

return <ReactGraphqlUIContext.Provider value={{
  defaultComponents: PrimeReactDefaultComponents,
  configsMap: HasuraConfig
}}>
  ...
</ReactGraphqlUIContext.Provider>
```

# Next Steps

If you're new to Tesseract's react-graphql library checkout:

* [How to setup your project](SetupYourProject.md)
* [How to configure react-graphql](reactGraphqlConfig.md)
* [Setting up your build](SettingUpYourBuild.md)
* [Dive deeper into Item vs Variable on mutations](ExplainingMutationItemVsVariable.md)

If you want to checkout the cool UI support we have look at

* [FlexForm](FlexForm.md)
* [FlexFormLocal](FlexFormLocal.md)
* [useDataTable](useDataTable.md)

Other references

* [Typescript support](ReactGraphqlTypescriptSupport.md)

> Third Parties we use

Reviewer: Missing react-hook-form I looked at their docs a handful of times while learning

Reviewer: Missing vite

* [PrimeReact](https://www.primefaces.org/primereact/showcase)
* [Jotai](https://jotai.pmnd.rs/docs/introduction)

> * [Tailwindcss](https://tailwindcss.com)

Reviewer: FYI when I was talking to Josh he'd mentioned PrimeFlex, not sure if he's had a chance to talk to you about that
